studio/hooks/analytics/useLogsPreview.tsx

```diff
@@ -56,7 +56,7 @@ function useLogsPreview(
   })
 
   useEffect(() => {
-    if (filters !== {}) {
+    if (Object.keys(filters).length !== 0) {
```

Comment:
seems like having this `if` will affect filter clearing. Could you remove the if statement instead? that way, we perform a refresh whenever the filters change.

Reply:
just updated. I removed this condition but adding a new one that will check and prevent the refresh to call from first render and will run if filters change. @Ziinc

Reply:
Seems like the tests do not pass, likely due to the `params.sql` null check in the useSWR key generation callback.
What is happening is that the sql parameter generation does not occur unless the `refresh` function is called.

I've also checked that there is only one data fetching requests being made in the current state, so the current change would result in regressions. Although intuitively, your change to performing the refresh on 2nd render makes sense, it seems that the that is not the case in tests due to the additional checks/deduping performed.

I can merge in the equality check removal first. If you'd like to dig into this further, I can recommend the following changes:
- generate the initial `params.sql` on hook init, currently an empty string, OR
- trigger the sql generation on first render.

I would also recommend running tests locally to verify that everything still works as expected.

Reply:
Thank you for your recommendation. Just updated, could you check?

Reply:
Hi, could you pull out the if condition first for this PR, and have the first render check update in a separate PR?